ultrasonic/src/main/kotlin/org/moire/ultrasonic/playback/CachedDataSource.kt

```diff
@@ -186,6 +186,8 @@ class CachedDataSource(
             transferEnded()
             responseByteStream?.close()
             responseByteStream = null
+        } else {
+            upstreamDataSource.close()
```

Comment:
But the real culprit was that the SSL connection wasn't closed in the underlying ApiDataSorce.
I haven't tested this fully but now the freezing and the timeouts seem to be gone.

Reply:
Wow so proud you found it 😊!